File: Solutions/0219- Contains Duplicate II/Solution.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        map = dict()

        for i in range(0,len(nums)):
            item = nums[i]
            if item not in map:
                map[item] = i
            else:
                if abs(map[item] - i) <= k:
                    return True
                else:
                    map[item] = i

        return False

File: Solutions/0219- Contains Duplicate II/Solution.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public boolean containsNearbyDuplicate(int[] nums, int k) {
        Map<Integer, Integer> map = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            int item = nums[i];
            if (!map.containsKey(item))
                map.put(item,i);
            else {
                if(Math.abs(map.get(item) - i) <= k)
                    return true;
                else
                    map.put(item, i);
            }
        }

        return false;
    }
}